BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,278 @@
+        help='Directory where universal binary will be stored',
+        default=default_conf["dst_app"])
+
+    parser.add_argument(
+        '--entitlements',
+        help='Path to .entitlements file for code signing',
+        default=default_conf["entitlements"])
+
+    parser.add_argument(
+        '--codesign',
+        help='Code signing identity to use to sign the applications',
+        default=default_conf["codesign_identity"],
+        dest="codesign_identity")
+
+    for arch in ARCHITECTURES:
+        parser.add_argument(
+             '--{}_pkg_config'.format(arch),
+             help="Folder containing .pc files for {} libraries".format(arch),
+             default=default_conf[arch+"_pkg_config_path"],
+             dest=arch+"_pkg_config_path")
+
+        parser.add_argument(
+             '--{}_qt5_path'.format(arch),
+             help="Install path for {} qt5 libraries".format(arch),
+             default=default_conf[arch+"_qt5_path"])
+
+        parser.add_argument(
+             '--{}_mac_os_deployment_target'.format(arch),
+             help="Deployment architecture for {} slice".format(arch),
+             default=default_conf[arch+"_mac_os_deployment_target"])
+
+    return vars(parser.parse_args())
+
+
+def lipo(path0, path1, dst):
+    if subprocess.call(['lipo', '-create', '-output', dst, path0, path1]) != 0:
+        print("WARNING: {} and {} can not be lipo'd, keeping {}"
+              .format(path0, path1, path0))
+
+        shutil.copy(path0, dst)
+
+
+def recursiveMergeBinaries(src0, src1, dst):
+    """
+    Merges two build trees together for different architectures into a single
+    universal binary.
+
+    The rules for merging are:
+
+    1) Files that exist in either src tree are copied into the dst tree
+    2) Files that exist in both trees and are identical are copied over
+       unmodified
+    3) Files that exist in both trees and are non-identical are lipo'd
+    4) Symlinks are created in the destination tree to mirror the hierarchy in
+       the source trees
+    """
+
+    # loop over all files in src0
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0)
+        newpath1 = os.path.join(src1, filename)
+        new_dst_path = os.path.join(dst, filename)
+        if os.path.islink(newpath0):
+            # symlinks will be fixed after files are resolved
+            continue
+
+        if not os.path.exists(newpath1):
+            # copy files that don't exist in path1
+            shutil.copy(newpath0, new_dst_path)
+            continue
+
+        if os.path.isdir(newpath1):
+            os.mkdir(new_dst_path)
+            # recurse into directories
+            recursiveMergeBinaries(newpath0, newpath1, new_dst_path)
+            continue
+
+        if filecmp.cmp(newpath0, newpath1):
+            # copy files that are the same
+            shutil.copy(newpath0, new_dst_path)
+        else:
+            # lipo together files that are different
```

Comment:
Ditto, regarding the comment meaningfulness.

Reply:
Okay, I can remove it.